backoffice/common/components/Pagination.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import ReactPaginate from 'react-paginate';
+import { Form } from 'react-bootstrap';
+import { PAGE_SIZE_OPTION } from '@commonServices/PaginationService';
+
+interface PaginationControls {
+  itemsPerPage?: {
+    value: number;
+    onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+  };
+  goToPage?: {
+    value: string;
+    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    onSubmit: () => void;
+  };
+}
+
+interface PaginationProps {
+  pageNo: number;
+  totalPage: number;
+  paginationControls?: PaginationControls;
+  onPageChange: (selectedItem: { selected: number }) => void;
+  pageSizeOption?: number[];
+}
+
+function Pagination({
+  pageNo,
+  totalPage,
+  paginationControls,
+  onPageChange,
+  pageSizeOption = PAGE_SIZE_OPTION,
+}: Readonly<PaginationProps>) {
+  return (
+    <div>
+      <ReactPaginate
+        forcePage={pageNo}
+        previousLabel="Previous"
+        nextLabel="Next"
+        pageCount={totalPage}
+        onPageChange={onPageChange}
+        containerClassName="pagination-container"
+        previousClassName="previous-btn"
+        nextClassName="next-btn"
+        disabledClassName="pagination-disabled"
+        activeClassName="pagination-active"
+      />
+      {paginationControls && (
+        <div className="pagination-helper mt-3 mb-3">
+          {paginationControls.goToPage && (
+            <div className={`pagination-tool ${paginationControls.itemsPerPage ? 'me-5' : ''}`}>
+              <p>Go to</p>
+              <Form.Control
+                type="number"
+                value={paginationControls.goToPage.value || ''}
+                onChange={paginationControls.goToPage.onChange}
+                onKeyDown={(e) => e.key === 'Enter' && paginationControls.goToPage?.onSubmit()}
```

Comment:
Hi @HnKnA, it looks like you're already checking for `paginationControls.goToPage`, so using the optional chaining operator `?.` with `paginationControls.goToPage?.onSubmit()` might not be needed here. You could consider simplifying it for consistency.

Reply:
The current check is needed because while I wrap paginationControls && paginationControls.goToPage in an existence check higher up:

```
{paginationControls.goToPage && (
 
)}
```

Inside the `onKeyDown` function, TypeScript **doesn't carry over** the knowledge that `goToPage` is defined.

- TypeScript only checks local scope for guarantees. It doesn't "remember" the checks made in the surrounding JSX block.
- To TypeScript, `paginationControls.goToPage` could have become `undefined` by the time `onKeyDown` runs.
- For `onChange`, React safely ignores `undefined` handlers, so even if `paginationControls.goToPage` is `undefined`, React won’t call it, and there’s no runtime risk.